python/pyproject.toml

```diff
@@ -20,7 +21,8 @@ classifiers = [
     "Operating System :: OS Independent",
 ]
 dependencies = [
-  "pyarrow~=17.0",
+  "pyarrow~=18.0",
+  "numpy>=1.16.6",
```

Comment:
We shouldn't need to have numpy as a runtime dependency of the Python package right?